mockwebserver/src/test/java/okhttp3/mockwebserver/MockWebServerTest.java

```diff
@@ -452,6 +455,24 @@
     assertEquals("foo bar", requestUrl.queryParameter("key"));
   }
 
+  @Test public void shutdownServerAfterRequest() throws Exception {
+    server.enqueue(new MockResponse().setSocketPolicy(SocketPolicy.SHUTDOWN_SERVER_AFTER_RESPONSE));
+
+    URL url = server.url("/").url();
+
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    assertEquals(HttpURLConnection.HTTP_OK, connection.getResponseCode());
+
+    HttpURLConnection refusedConnection = (HttpURLConnection) url.openConnection();
+
+    try {
+      refusedConnection.getResponseCode();
+      fail("Second connection should be refused");
+    } catch (ConnectException e ) {
+      assertThat(e, hasMessage(containsString("refused")));
```

Comment:
We don't use hamcrest anywhere. If we adopted something new it'd probably be Truth. Perhaps just stick to `assertTrue(e.getMessage().contains("refused"))` for now.

Reply:
fixed